internal/common/typescript_mock_lib.bzl

```diff
@@ -19,19 +19,42 @@ rules_typescript without introducing a circular dependency between
 rules_nodejs and rules_typescript repositories.
 """
 
+load("@build_bazel_rules_nodejs//:declaration_provider.bzl", "DeclarationInfo")
+load("@build_bazel_rules_nodejs//:providers.bzl", "JSEcmaScriptModuleInfo")
+
 def _mock_typescript_lib(ctx):
-    es5_sources = depset()
-    transitive_decls = depset()
-    for s in ctx.attr.srcs:
-        files_list = s.files.to_list()
-        es5_sources = depset([f for f in files_list if f.path.endswith(".js")], transitive = [es5_sources])
-        transitive_decls = depset([f for f in files_list if f.path.endswith(".d.ts")], transitive = [transitive_decls])
+    es5_sources = depset([f for f in ctx.files.srcs if f.path.endswith(".js")])
+    es6_sources = depset([f for f in ctx.files.srcs if f.path.endswith(".mjs")])
+    declarations = depset([f for f in ctx.files.srcs if f.path.endswith(".d.ts")])
     return struct(
-        runfiles = ctx.runfiles(collect_default = True, collect_data = True),
         typescript = struct(
+            declarations = declarations,
+            devmode_manifest = None,
             es5_sources = es5_sources,
-            transitive_declarations = transitive_decls,
+            es6_sources = es6_sources,
+            replay_params = None,
+            transitive_declarations = declarations,
+            transitive_es5_sources = es5_sources,
+            transitive_es6_sources = es6_sources,
+            tsickle_externs = [],
+            type_blacklisted_declarations = depset(),
         ),
+        providers = [
```

Comment:
feels like it just wants to be a real thing

unless we want this to be "produces_jsprovider.bzl" and drop all references to typescript

Reply:
I'll rip out mock_typescript_lib in a pre-factor PR

Reply:
Done in #1212